### PR TITLE
Validate outcome result score

### LINKED ISSUE #72: Outcomes result score isn't validated
The outcomes results score is required and supposed to be a decimal between 0 and 1.0, but there's no validation on a replaceResultRequest, so you can pass in anything.

### PR DESCRIPTION
**Pull request by brainiac10**
Fixes #72

### CODE REVIEW

**Review by andyfmiller — DISMISSED** [review]
After writing this, I realized I could more easily add comments inline, but I can't remove this review. I'll use inline comments next time.

File: src/LtiLibrary.AspNetCore/Outcomes/v1/OutcomesControllerBase.cs [hunk 1]
```diff
@@ -261,6 +261,16 @@ private static Result GetResult(ResultRecordType resultRecord)
             var replaceResponse = new replaceResultResponse();
 
             var result = GetResult(replaceRequest.resultRecord);
+
+            //The LTI 1.1 specification states in 6.1.1:
+            //The TC must check the incoming grade for validity and must fail when a grade is outside the range 0.0-1.0 or if the grade is not a valid number.   
+            //The TC must respond to these replaceResult operations with a imsx_codeMajor of "failure".
```
Comment (andyfmiller) [inline src/LtiLibrary.AspNetCore/Outcomes/v1/OutcomesControllerBase.cs:267]: Would you mind referencing the URL https://www.imsglobal.org/specs/ltiv1p1/implementation-guide#toc-26

File: test/LtiLibrary.AspNetCore.Tests/Outcomes/v1/OutcomesControllerShould.cs [hunk 2]
```diff
@@ -70,6 +70,17 @@ public OutcomesControllerShould()
             Assert.True(replaceResult.StatusCode == HttpStatusCode.Unauthorized, $"{replaceResult.StatusCode} == {HttpStatusCode.Unauthorized}");
         }
 
+        [Theory]
+        [InlineData(null)]
+        [InlineData(-.01)]
+        [InlineData(1.01)]
+        public async void NotReplaceResult_IfScoreIsInvalid(double? score)
+        {
+            // First verify that plumbing works if secret is correct
```
Comment (andyfmiller) [inline test/LtiLibrary.AspNetCore.Tests/Outcomes/v1/OutcomesControllerShould.cs:79]: This comment does not describe what is about to happen. Copy and paste artifact?

Comment (brainiac10) [reply test/LtiLibrary.AspNetCore.Tests/Outcomes/v1/OutcomesControllerShould.cs:79]: Whoops, yep.  Copied from the adjacent test.

**Comment by andyfmiller** [comment]
I merged the PR and pushed the two minor changes to master.

Thank you for finding and fixing that issue! And for showing me a better way to right a test (I've never tried [Theory] before).

**Comment by brainiac10** [comment]
No problem at all, I just made a tweak.  You did all the heavy lifting.  :)

**Pull request merged**